state/protocol/protocol_state/epochs/statemachine.go

```diff
@@ -271,6 +275,18 @@ func (e *EpochStateMachine) applyServiceEventsFromOrderedResults(orderedUpdates
 			if processed {
 				dbUpdates.AddDbOp(e.commits.StoreTx(ev)) // we'll insert the commit event when we insert the block
 			}
+		case *flow.EpochRecover:
+			processed, err := e.activeStateMachine.ProcessEpochRecover(ev)
+			if err != nil {
+				if protocol.IsInvalidServiceEventError(err) {
+					// TODO for 'leaving Epoch Fallback via special service event': add reporting of invalid service event, at least a log.
+					continue
```

Comment:
My intuition is to not swallow an `InvalidServiceEventError` here:
* On the happy path, if we receive an `EpochRecover` event, do we just stay on the happy path and ignore this event? 

  I feel this would be inconsistent with our approach of "everything unexpected results in EFM".

  So if we just escalated this error here as any other error during processing, we would enter EFM 

* If we are already in EFM, I think the `FallbackStateMachine` should handle this internally. See my explanation in https://github.com/onflow/flow-go/issues/6018. Essentially, my mindset is that the `FallbackStateMachine` should more or less tolerate all events without _any_ errors, unless it detects symptoms of an internal bug within the protocol state. 

This PR is already quite big. I would be inclined to address this as part of issue https://github.com/onflow/flow-go/issues/6018